examples/occlusion.html

```diff
@@ -0,0 +1,392 @@
+
+  // Control the resizing of the window to correctly display the scene.
+  window.addEventListener( 'resize', onWindowResize, false );
+  // Wherever the user clicks in the screen, place the model.
+  renderer.domElement.addEventListener("click", function(event) {
+    pos.x = event.pageX / window.innerWidth;
+    pos.y = event.pageY / window.innerHeight;
+    picking();
+  });
+  // Control the rotation of the model
+  var lastX = 0;
+  renderer.domElement.addEventListener("touchstart", function(event) {
+    if (event.changedTouches.length > 0)
+      lastX = event.changedTouches[0].pageX;
+  });
+  renderer.domElement.addEventListener("touchmove", function(event) {
+      if (event.changedTouches.length > 0 && vrDisplay) {
+        var x = event.changedTouches[0].pageX;
+        var diffX = x - lastX;
+        lastX = x;
+        model.rotateX(THREE.Math.degToRad(diffX));
+      }
+  });
+
+  update();
+}
+function picking() {
+  if (vrDisplay) {
+    var hits =
+        vrDisplay.hitTest(pos.x, pos.y);
+    if (hits && hits.length > 0) {
+      var hit = hits[0];
+      // Orient the cone model at the position and orientation of the hit.
+      THREE.ARUtils.placeObjectAtHit(model, hit, 1, true);
+      points.material.uniforms.plane.value.fromArray(
+          [hit.modelMatrix[4], hit.modelMatrix[5], hit.modelMatrix[6], hit.modelMatrix[7]]);
+      /*
+      // Orient and position the model in the picking point according
+      // to the picking plane. The offset is half of the model size.
+      THREE.WebAR.
+        positionAndRotateObject3DWithPickingPointAndPlaneInPointCloud(
+          pointAndPlane, model, MODEL_SIZE_IN_METERS / 2);
+      // Provide the point material with the plane so it can discard
+      // the points that are considered to be in the plane.
+      points.material.uniforms.plane.value.fromArray(
+        pointAndPlane.plane);
+    */
+    }
+    else {
+      console.warn("Could not retrieve the correct point and plane.");
+    }
+  }
+}
+
+function update() {
+  // Render the device's camera stream on screen first of all.
+  // It allows to get the right pose synchronized with the right frame.
+  arView.render();
+
+  // Update our camera projection matrix in the event that
+  // the near or far planes have updated
+  camera.updateProjectionMatrix();
+
+  // Update our perspective camera's positioning
+  vrControls.update();
+
+  // Render our three.js virtual scene
+  renderer.clearDepth();
+
+  // If continuous picking is enabled, use the center of the screen
+  // to continuously pick.
+  if (gui.continuousPicking) {
+    pos.x = 0.5;
+    pos.y = 0.5;
+    picking();
+  }
+
+  // Update the point cloud. Only if the point cloud will be shown the
+  // geometry is also updated.
+  pointCloud.update(gui.usePointCloudForDepth, 0, gui.transformPoints);
+
+  // Render the point cloud
+  if (this.gui.usePointCloudForDepth) {
+    this.renderer.context.colorMask(false, false, false, false);
```

Comment:
This is where the occlusion magic happens -- some extra comments here would be helpful as a standard example for occlusion in web AR

Reply:
done